services/visual-identity/api.go

```diff
@@ -0,0 +1,92 @@
+package visualidentity
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/static"
+)
+
+const (
+	emojiAlphabetLen       = 2757 // 20bytes of data described by 14 emojis requires at least 2757 length alphabet
+	emojiHashLen           = 14
+	colorHashSegmentMaxLen = 5
+	colorHashColorsCount   = 32
+)
+
+func NewAPI() *API {
+	colorHashAlphabet := MakeColorHashAlphabet(colorHashSegmentMaxLen, colorHashColorsCount)
+	return &API{
+		emojisAlphabet:    &[]string{},
+		colorHashAlphabet: &colorHashAlphabet,
+	}
+}
+
+type API struct {
+	emojisAlphabet    *[]string
+	colorHashAlphabet *[][]int
+}
+
+func (api *API) EmojiHashOf(pubkey string) (hash []string, err error) {
+	log.Info("[VisualIdentityAPI::EmojiHashOf]")
```

Comment:
Probably we can change this to .Debug or .Trace (if it exists), since it might pollute the logs too much